src/textual_fspicker/select_directory.py

```diff
@@ -91,23 +61,61 @@ def __init__(
 
     def on_mount(self) -> None:
         """Configure the dialog once the DOM is ready."""
+        super().on_mount() # Call parent's on_mount
         navigation = self.query_one(DirectoryNavigation)
         navigation.show_files = False
-        self.query_one(CurrentDirectory).current_directory = navigation.location
+
+        path_input = self.query_one("#path_input", Input)
+        path_input.value = str(navigation.location)
+        # navigation.focus() # Focus is handled by super().on_mount or should be reconsidered
 
     def _input_bar(self) -> ComposeResult:
         """Provide any widgets for the input before, before the buttons."""
-        yield CurrentDirectory()
+        yield Input(id="path_input", placeholder="Type path or select below")
 
     @on(DirectoryNavigation.Changed)
-    def _show_selected(self, event: DirectoryNavigation.Changed) -> None:
-        """Update the display of the current location.
+    def _update_path_input_on_nav_change(self, event: DirectoryNavigation.Changed) -> None:
+        """Update the display of the current location in the Input widget.
 
         Args:
             event: The event with the selection information in.
         """
+        # This combines the original _show_selected from SelectDirectory
```

Comment:
Comments about how code used to work aren't really that helpful. I'd recommend placing this sort of information in the description of the commit that makes the change, and more general in a rationale for the approach of a PR, as written in the description.